templates/digital-asset-template/frontend/hooks/useGetCollections.tsx

```diff
@@ -1,33 +1,24 @@
 import { aptosClient } from "@/utils/aptosClient";
-import { AccountAddress, GetCollectionDataResponse } from "@aptos-labs/ts-sdk";
-import { useState, useEffect } from "react";
+import { AccountAddress } from "@aptos-labs/ts-sdk";
+import { useQuery } from "@tanstack/react-query";
 
 /**
  * A react hook to get all collections under the current contract.
  *
  * This call can be pretty expensive when fetching a big number of collections,
  * therefore it is not recommended to use it in production
- *
  */
 export function useGetCollections() {
-  const [collections, setCollections] = useState<
-    Array<GetCollectionDataResponse>
-  >([]);
-
-  useEffect(() => {
-    // fetch the contract registry address
-    getRegistry().then((registry) => {
-      // fetch collections objects created under that contract registry address
-      getObjects(registry).then((objects) => {
-        // get each collection object data
-        getCollections(objects).then((data) => {
-          setCollections(data);
-        });
-      });
-    });
-  }, []);
-
-  return collections;
+  return useQuery({
+    queryKey: ["collections"],
+    refetchInterval: 10 * 60 * 1000, // 10 sec
```

Comment:
does useQuery handle the refetch interval when data has been fetched? i.e if we have data, will it try to refetch?

Reply:
We could configure it to do so, however I think it is better to just let it refetch indefinitely. 

In the case we are minting for the first time and we go from 0 collections to 1 collection, this would be fine. However going from 1 to 2+ means we wouldn't know (easily at least) when we are done and the indexer is caught up. 

We could go with a different approach by polling the indexer if you want.

Reply:
My concern is about rate limit , we query indexer/node testnet/mainnet and with repeating query we can easily reach rate limit. We could resolve it by introducing api key but would probably work on it later.